### PR TITLE
Upgrade to Go 1.9beta2

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

Bumps Go version to v1.9 beta to see if there is anything that breaks

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

related to: #49484 

**Special notes for your reviewer**:
We might not want to merge this before in the v1.9 cycle. Yet to be decided though.
I have no idea if the bazel `WORKSPACE` syntax is correct, but I hope so :smile:

**Release note**:

```release-note
Upgrade Go version to v1.9beta2
```

cc @kubernetes/sig-cluster-lifecycle-pr-reviews  @kubernetes/sig-testing-pr-reviews @kubernetes/sig-release-pr-reviews @bradfitz @jessfraz @kubernetes/sig-scalability-pr-reviews

@ixdy Could you please push the image so that we can run the presubmit tests?

### CODE REVIEW

**Comment by cblecker** [comment]
There's a pending update to kube-cross here: https://github.com/kubernetes/kubernetes/pull/46193

It would be great to get those changes in before this so that the updated 1.9 image contains both changes.

**Comment by luxas** [comment]
@cblecker sounds good to me.
Notify me when I can rebase and @ixdy can push

**Comment by cblecker** [comment]
@luxas As the bot mentioned, that PR merged last night so you are clear to rebase. Additionally, since then go1.9rc1 dropped so we could try with that :)

**Comment by luxas** [comment]
@ixdy @cblecker Updated

File: build/build-image/cross/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-v1.8.3-2
+v1.9.0rc1-1
```
Comment (cblecker) [inline build/build-image/cross/VERSION]: all the other references are to `1.9rc1`, while this one is `1.9.0rc1`. is this intended, @luxas?

**Review by cblecker — COMMENTED** [review]
One question, otherwise this looks good to test :)

File: build/build-image/cross/VERSION (hunk 1 above)
Comment (luxas) [inline build/build-image/cross/VERSION]: yes, it is intended. Seems like we've consistently used `vX.Y.Z-N` here, so this probably sorts/lists better

Comment (cblecker) [inline build/build-image/cross/VERSION]: Works for me!

**Comment by luxas** [comment]
@ixdy would you mind pushing the required image so we can see if this works?

File: build/build-image/cross/VERSION (hunk 1 above)
Comment (ixdy) [inline build/build-image/cross/VERSION]: there's an rc2 now. can you bump?

File: build/root/WORKSPACE [hunk 2]
```diff
@@ -43,7 +43,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_repositories")
 load("@io_bazel_rules_docker//docker:docker.bzl", "docker_repositories", "docker_pull")
 
 go_repositories(
-    go_version = "1.8.3",
+    go_version = "1.9rc1",
```
Comment (ixdy) [inline build/root/WORKSPACE]: The bazel go rules don't know about the RCs, unfortunately.

**Comment by cblecker** [comment]
@luxas as go1.9 is GA now, maybe we can try this again?

**Comment by luxas** [comment]
@cblecker I'm gonna be OOO for some time; could you please take it over?

**Comment by cblecker** [comment]
@luxas yeah, no problem.

**Comment by luxas** [comment]
@cblecker Thanks!

**Comment by dims** [comment]
oooo shiny!

**Comment by cblecker** [comment]
Opened https://github.com/kubernetes/kubernetes/pull/51375 :)
/close

**Pull request closed without merging**